mantidimaging/core/io/csv_output.py

```diff
@@ -11,18 +11,23 @@ class CSVOutput:
     def __init__(self) -> None:
         self.columns: dict[str, np.ndarray] = {}
         self.num_rows: int | None = None
+        self.units: dict[str, str] = {}
 
-    def add_column(self, name: str, values: np.ndarray) -> None:
+    def add_column(self, name: str, values: np.ndarray, units: str | None = None) -> None:
```

Comment:
If you made the units be required, and passed in `Counts` when calling it for the counts columns, then `write()` would be simpler.

I think it would be good practice to always write a units line.